src/main/java/com/arpnetworking/metrics/mad/model/json/Version2g.java

```diff
@@ -0,0 +1,394 @@
+            }
+
+            /**
+             * Sets the value field.
+             *
+             * @param value Value
+             * @return This builder
+             */
+            public Builder setValue(final Double value) {
+                _value = value;
+                return this;
+            }
+
+            /**
+             * Sets the compound unit field.
+             *
+             * @param value Value
+             * @return This builder
+             */
+            public Builder setUnit(final Unit value) {
+                _unit = value;
+                return this;
+            }
+
+            @NotNull
+            private Double _value;
+            private Unit _unit;
+        }
+
+        /**
+         * 2G format Unit class.
+         */
+        public static final class Unit {
+
+            public List<CompositeUnit> getNumerators() {
+                return _numerators;
+            }
+
+            public List<CompositeUnit> getDenominators() {
+                return _denominators;
+            }
+
+            private Unit(final Builder builder) {
+                _numerators = builder._numerators == null
+                        ? ImmutableList.of()
+                        : ImmutableList.copyOf(builder._numerators);
+                _denominators = builder._denominators == null
+                        ? ImmutableList.of()
+                        : ImmutableList.copyOf(builder._denominators);
+            }
+
+            private final List<CompositeUnit> _numerators;
+            private final List<CompositeUnit> _denominators;
+
+            /**
+             * Builder for the 2G Unit class.
+             */
+            public static class Builder extends OvalBuilder<Unit> {
+                /**
+                 * Public constructor.
+                 */
+                public Builder() {
+                    super(Unit::new);
+                }
+                /**
+                 * Sets the unit numerators field.
+                 *
+                 * @param value Value
+                 * @return This builder
+                 */
+                public Builder setNumerators(final List<CompositeUnit> value) {
+                    _numerators = value;
+                    return this;
+                }
+
+                /**
+                 * Sets the unit denominators field.
+                 *
+                 * @param value Value
+                 * @return This builder
+                 */
+                public Builder setDenominators(final List<CompositeUnit> value) {
+                    _denominators = value;
+                    return this;
+                }
+
+                private List<CompositeUnit> _numerators;
+                private List<CompositeUnit> _denominators;
+            }
+        }
+    }
+
+    /**
+     * Represents a counter, timer, or gauge element.  Includes a list of
+     * samples.
+     */
+    public static final class Element {
```

Comment:
Why is unit inside sample and sample not inside element? My preference is to not nest these classes within each other.